[NEW] weave-cloud

errordeveloper opened the pull request.

errordeveloper committed:
[NEW] weave-cloud

errordeveloper commented on stable/weave-cloud/Chart.yaml:
```diff
@@ -0,0 +1,11 @@
+name: weave-cloud
+version: latest
```
Obviously this doesn't work –`[ERROR] Chart.yaml: version 'latest' is not a valid SemVer`.
What is the best practice for versioning a chart that consists of multiple components?
Is it okay to call it `1.0.0` to start with?

lachie83 replied:
Start with 0.1.0 please

errordeveloper replied:
Updated.

errordeveloper commented:
cc @rimusz

errordeveloper commented:
ping @prydonius @viglesiasce

errordeveloper commented:
@lachie83 any chance you could take another look at this?

rimusz commented:
LGTM

lukemarsden commented:
Can we get this merged?

lachie83 commented on stable/weave-cloud/Chart.yaml:
```diff
@@ -0,0 +1,11 @@
+maintainers:
+  - name: Ilya Dmitrichenko
+    email: ilya@weave.works
+engine: gotpl
```
Would you like to add icon so that it shows up pretty in kubeapps.com?

errordeveloper replied:
What types of images are supported here?

viglesiasce replied:
I believe only png (as thats the only one that I've seen used).

@migmartri can you comment on what kind of images are supported on the Monocular/KubeApps side?

migmartri replied:
> @migmartri can you comment on what kind of images are supported on the Monocular/KubeApps side?

I believe that we currently support png, jpeg, gif, tiff and bpm.

errordeveloper replied:
I will add an icon once we agreed on all other threads in this PR.

lachie83 commented on stable/weave-cloud/templates/manifest.yaml:
```diff
@@ -0,0 +1,402 @@
+apiVersion: v1
+kind: List
```
I find this hard to review as a single list. Would you mind splitting into individual manifests?

errordeveloper replied:
As I said, this is a copy of manifest generated by our API, so it would be very impractical for us to split this up into very small pieces, but we can split them up into 3 pieces (one per agent, i.e. Flux, Scope & Cortex), if that would help with review...

lachie83 replied:
Understood

lachie83 commented on stable/weave-cloud/templates/manifest.yaml:
```diff
@@ -0,0 +1,402 @@
+items:
+  - metadata:
+      labels:
+        name: weave-flux
```
Please template the name

lachie83 replied:
Is the expectation that this can only be installed once per namespace per cluster?

errordeveloper replied:
Once per-cluster really.

lachie83 commented on stable/weave-cloud/templates/manifest.yaml:
```diff
@@ -0,0 +1,402 @@
+        name: weave-flux
+      name: weave-flux
+      annotations:
+        cloud.weave.works/launcher-info: |-
```
What are these annotations and should they be hardcoded? Also would it help to put them in a file and include the file as they seem to be used on all resources.

errordeveloper replied:
These are produced by our generator API, and are designed to help us supporting our users, so we can tell exactly when user installed the resources and how far are they behind latest. I am happy to modify the content of this annotation for the purpose of this chart.

lachie83 commented on stable/weave-cloud/templates/manifest.yaml:
```diff
@@ -0,0 +1,402 @@
+        spec:
+          containers:
+            - name: agent
+              image: 'quay.io/weaveworks/fluxd:0.1.0'
```
template all resources and include in values.yaml so that they can be changed.

errordeveloper replied:
We prefer to use our upstream manifest (maintained as an API see comment below regarding `update.sh`).
I am happy to apply some minimal transformation to the version that would get check-in here, but I would refrain from using templates for names of resources and would prefer to keep our well-known names instead.

> ...so that they can be changed.

Could you please explain what is the user benefit of that?

lachie83 replied:
This is fine. It just means updates to container images names and/or tags will result in a new chart version.

lachie83 commented on stable/weave-cloud/templates/manifest.yaml:
```diff
@@ -0,0 +1,402 @@
+            weave-scope-component: agent
+          annotations:
+            scheduler.alpha.kubernetes.io/tolerations: >-
+              [{"key":"dedicated","operator":"Equal","value":"master","effect":"NoSchedule"}]
```
Hard to read. Pease line break and comment

errordeveloper replied:
This one is standard, actually will change if we check-in 1.6 version of this manifest instead, which leads to another question – how are Kubernetes versions handled in charts repo? Our manifests look quite different between 1.5 and 1.6 (this one is 1.5).

lachie83 replied:
We only have a single branch at the moment. We have conditionalized pieces of the manifests in the past. It's something we need to figure out. Simply place the prereq version in the chart readme.

errordeveloper replied:
Ack.

lachie83 commented on stable/weave-cloud/templates/manifest.yaml:
```diff
@@ -0,0 +1,402 @@
+              args:
+                - '--no-app'
+                - '--probe.docker.bridge=docker0'
+                - '--probe.docker=true'
```
Can this probe any other container runtimes? Should these be configurable? What happens if I'm not using docker0?

errordeveloper replied:
Currently only Docker is supported. I am aware that in some setups the bridge interface may be something like `cbr0`, but we've not come across such setups recently.

lachie83 commented on stable/weave-cloud/README.md:
```diff
@@ -0,0 +1,40 @@
+
+Weave Cloud is a add-on to Kubernetes which provides Continuous Delivery, along with hosted Prometheus Monitoring and a visual dashboard for exploring & debugging microservices.
+
+This package contains the agents which connect your cluster to Weave Cloud.
```
Can you please add a little more details about what resources are exactly installed on the cluster?

errordeveloper replied:
Makes sense, I can do that.

lachie83 commented on stable/weave-cloud/update.sh:
```diff
@@ -0,0 +1,2 @@
+#!/bin/sh -eux
```
Provide a little more detail on the purpose of this

errordeveloper replied:
Glad someone is finally reviewing this!

What we have is this generator API that is normally used with `kubectl create -n kube-system https://cloud.weave.works/k8s.yaml?t=<service_token>`, but for this chart we simply check-in the YAML generated by the API, hence this script here.

lachie83 replied:
Thanks

lachie83 reviewed (changes requested):
Thanks for the contrib. Please review the technical requirements in order for this to be merged into stable.

lukemarsden commented:
Looks like there are some review comments for you to address, @errordeveloper 😉

errordeveloper commented:
@lachie83 thanks for reviewing this! Please reply to my comments and I'll make changes once we are clear.

errordeveloper commented:
@lachie83 NB: it's probably best to read my comments from bottom to top...

errordeveloper commented:
@lachie83 ping.

errordeveloper commented:
@lachie83 I've rebased and refactored this based on the comments you've made earlier.

- there are now 3 separate files for each of the components
- support annotations had been removed
- added a comment describing what `update.sh` is all about
- added an icon

This version should be also easier to review as we now have a more natural sorting order of the YAML key.

lachie83 commented on stable/weave-cloud/templates/NOTES.txt:
```diff
@@ -0,0 +1,9 @@
+Verify all Pods are running
```
Will this work without a service token? If not is it worth doing something like the following?

https://github.com/kubernetes/charts/blob/master/stable/sumokube/templates/NOTES.txt#L1

errordeveloper replied:
Thanks for the pointer! I think `required` function seems more suitable, I think `NOTES.txt` solution predates `required`...

errordeveloper commented:
It turns out that tests are failing because `helm lint` errors on `required` function in the templates, the only other chart that uses `required` without a default value is `stable/kube-lego`. The test status seem had been ignored when most recent PR (https://github.com/kubernetes/charts/pull/1064) was merged for that chart.

lachie83 commented:
@errordeveloper Thanks for the update. Just want to confirm that you're happy for this chart UX before we go to merge. I personally like the other way I mentioned so at least you get can through CI and the user is prompted to do an upgrade with the service token. Thoughts?

errordeveloper commented:
@lachie83 I agree that the UX is not very idea, to be honest I wish there was better way, but what you are suggesting makes the UX clearer, especially because it allows for nicer formatting of the error message.

lachie83 commented:
@errordeveloper Are you going to push a commit? I think that's it then we can merge

errordeveloper commented:
@lachie83 done!

lachie83 commented:
@errordeveloper if CI is happy then let's merge this thing

errordeveloper commented:
@lachie83 looks like it just passed!
